src/poli/core/util/external_observer.py

```diff
@@ -12,31 +13,67 @@ class ExternalObserver(AbstractObserver):
     User-defined observers typically do NOT inherit from here.
     """
 
-    def __init__(self, observer_name: str = None):
+    def __init__(self, observer_name: str = None, **kwargs_for_observer):
         if observer_name is None:
             observer_name = _DEFAULT
 
         self.observer_script = config[observer_name][_OBSERVER]
         self.process_wrapper = None
+        self.kwargs_for_observer = kwargs_for_observer
 
     def observe(self, x: np.ndarray, y: np.ndarray, context=None) -> None:
-        self.process_wrapper.send([x, y, context])
+        # We send the observation
+        self.process_wrapper.send(["OBSERVATION", x, y, context])
+
+        # And we make sure the process received and logged it correctly
+        msg_type, *msg = self.process_wrapper.recv()
+        if msg_type == "EXCEPTION":
+            raise msg[0]
```

Comment:
Lovely.

Reply:
Just FYI, the communication is a between `external_observer.py` and `observer_wrapper.py`.